### PR TITLE
Add connection_string secret to default_user secret

### LINKED ISSUE #1722: Add connection_string to the default_user secret
**Is your feature request related to a problem? Please describe.**
Currently the default_user secret has the connection credentials, but it does not have a formatted connection_string that is directly usable.
When using the operator in my code it is not a problem because I can build the connection_string myself, but I am trying to use KEDA to autoscale my pods based on the RabbitMQ queue state, and KEDA needs a connection_string to be able to connect to RabbitMQ, thus making it already available very valuable.

**Describe the solution you'd like**
Another key in the secret with the fully built connection_string.

### PR DESCRIPTION
**Pull request by hxyannay**
This closes #1722

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes
Added a new key `connection_string` to the default_user secret with a formatted connection_string.

### CODE REVIEW

File: internal/resource/default_user_secret_test.go
Comment (Zerpet) [inline internal/resource/default_user_secret_test.go:120]: This works, but it would read better by using the matcher `HaveKey` ([link](https://onsi.github.io/gomega/#havekeykey-interface)).

File: internal/resource/default_user_secret_test.go
Comment (Zerpet) [inline internal/resource/default_user_secret_test.go:212]: This can be re-written using the `HaveKeyWithValue` [matcher](https://onsi.github.io/gomega/#havekeywithvaluekey-interface-value-interface). I recon you took existing code and extended the test following the same pattern 🙂 But it is a good opportunity to refactor some old code.

**Review by Zerpet — COMMENTED** [review]
Thank you for this contribution. I like this idea. I left some cosmetic suggestions.

File: internal/resource/default_user_secret_test.go
Comment (hxyannay) [reply internal/resource/default_user_secret_test.go]: Great suggestion! I am not that familiar with go and the framework tests so I didn't know the most optimal way and I copied.
Did what you suggested, can you take another look?

File: internal/resource/default_user_secret_test.go
Comment (hxyannay) [reply internal/resource/default_user_secret_test.go]: Did this as well :)

**Comment by hxyannay** [comment]
Committed your changes :)
Also it would be great if you could release a new version of the operator once you merge this pull request, so I could use those changes right away.

**Comment by hxyannay** [comment]
Hey @Zerpet 
Can you take another look?
Thanks!

**Review by Zerpet — APPROVED** [review]

**Comment by Zerpet** [comment]
I approved the PR ✅ I'll merge after CI passes. We released 2.10.0 last week, and it is a bit soon to release another version. We tend to wait for 1 month between minor releases. Plus we have some dependabot PRs that need manual intervention.

If you want to use these changes right away, I'd recommend that you build own "snapshot" version from `main` after merging. There's a Make target to build, and it only requires 2 inputs, see:

https://github.com/rabbitmq/cluster-operator/blob/86e948edcaf430f78ff44de8860d0ab47a548247/Makefile#L207-L210

You can build the operator locally, for example, with: 

```
make docker-build DOCKER_REGISTRY_SERVER=docker.io OPERATOR_IMAGE=hxyannay/rabbitmq-cluster-operator
```

You will have to push the image then with `docker push` and update the latest release manifests to use this image.

**Comment by hxyannay** [comment]
Great, I will build an image myself for the time being. 
Thank you.

**Pull request merged**
